privacyidea/api/lib/prepolicy.py

```diff
@@ -151,6 +152,9 @@ def init_random_pin(request=None, action=None):
                                                unique=True,
                                                audit_data=g.audit_object.audit_data)
 
+    if action=="setrandompin" and len(pin_pols) == 0:
```

Comment:
I think we might have a logic problem here: the otp_pin_random policy current generates a random PIN during the enrollment of a token. 

This might be problematic: If we want to set a random pin only manually, the configuration/policy will also set a random pin during /token/init, which might not be the desired behaviour.

What I think woud actually be the right way would be to remove the decorator "init_random_pin" from /token/init and only have the policy otppinrandom for the /token/setrandompin endpoint *and* for the token handler.

The /token/init random PIN could than be handled by the tokenhandler.

I think it is important to split the /token/init and the /token/setrandompin in regards to actually create a random pin. Since as mentioned above.

What do *you* think?

Reply:
Or we should use a 2nd decorator and a 2nd policy.

Reply:
I did this in 7d23bd01

Reply:
Thanks, I like the solution with a second policy.